File: factorial.py
"""
    Calculate a factorial using recursion
"""

def factorial(integer):
    for i in range(1, integer+1):
        if i>1:
            result = i*factorial(i-1)
        else:
            result = 1
    return result


print("Enter a number whose factorial is needed: ")
n = int(input())
print(factorial(n))

File: factorial.cpp
/*
 *	CPP program to find the factorial of a given number 
 */
#include <iostream>
int main(){
	int N;
	double fact = 1;
	std::cout << "Enter a number: ";
	std::cin >> N;

	for(int i =1; i<=N; i++){
		fact = fact*i;
	}
	
	std::cout << "The Factorial of " << N << " is: " << fact << std::endl;


}
